Account balance display for vesting accounts

atharen opened the pull request:
- Introduced spendable and locked balance calculations for vesting accounts.
- Updated UI to differentiate between total, spendable, and locked balances.
- Added logic to detect vesting accounts and load corresponding spendable balances.
- Improved overall balance display for both vesting and regular accounts.

atharen committed:
Account balance display for vesting accounts

- Introduced spendable and locked balance calculations for vesting accounts.
- Updated UI to differentiate between total, spendable, and locked balances.
- Added logic to detect vesting accounts and load corresponding spendable balances.
- Improved overall balance display for both vesting and regular accounts.

atharen committed:
Enhance vesting account balance display

- Refined balance calculations for vesting accounts to include liquid and locked balances.
- Updated UI to reflect new balance categories: Liquid, Vesting, Delegation, Reward, and Unbonding.
- Added theoretical vested amount calculation based on vesting schedule.
- Improved clarity of balance descriptions and tooltips for better user understanding.

migueldingli1997 commented on src/modules/[chain]/account/[address].vue:
```diff
@@ -194,36 +323,182 @@ function mapAmount(events:{type: string, attributes: {key: string, value: string
+                    class="absolute top-0 bottom-0 left-0 right-0 bg-info opacity-20"
+                  ></div>
+                </div>
+                <div class="flex-1">
+                  <div class="text-sm font-semibold">
+                    {{ format.formatToken({
+                      amount: String(theoreticalVestedAmount),
+                      denom: balanceItem.denom
+                    }) }} <span class="text-xs text-info">Vested (Unlocked)</span>
+                  </div>
+                  <div class="text-xs text-info">
+                    Made spendable by vesting schedule, may have been utilized
+                  </div>
+                </div>
+                <div
+                  class="text-xs truncate relative py-1 px-3 rounded-full w-fit text-primary dark:invert mr-2"
+                >
+                  <span
+                    class="inset-x-0 inset-y-0 opacity-10 absolute bg-primary dark:invert text-sm"
+                  ></span>
+                  ${{ format.tokenValue({
+                      amount: String(theoreticalVestedAmount),
+                      denom: balanceItem.denom
+                    }) }}                
+                </div>
+              </div>
```
Why do values appear twice (I see `theoreticalVestedAmount` twice here).

atharen replied:
The first is the actual token formatted, the second is the value (USD) formatted.

_Standard for other balances as well._

atharen committed:
feat(account): Refactor vesting account handling

- handling multi-vesting accounts with multiple schedule
- UI to display vesting schedule information, including progress.
- Refined balance display
- Added helper functions for better management of vesting account data.

atharen committed:
feat(account): Improve vesting account balances

- Balance calculations to leverages spendable and vesting amounts.
- Enhanced UI labels for clarity:
- 'Locked' to 'Vesting'
- 'Liquid' to 'Balance/Spendable'.
- Simplified balance rendering for maintainability.

atharen committed:
feat(chain): Update fuel mainnet configuration

- Reformatted JSON structure for better readability.
- Added asset details for FUEL including coingecko_id and logo.
- Specified features supported by the fuel mainnet.
- Removed redundant asset entry and ensured consistent formatting.

atharen committed:
readme: changelog for vesting account changes

changes related to support for vesting accounts and spendable balances

atharen committed:
fixup! feat(account): Improve vesting account balances

atharen committed:
fixup! fixup! feat(account): Improve vesting account balances

atharen commented:
Replaced by #8
